### DISCUSSION TITLE
Hide error stacktrace

### DISCUSSION THREAD

**Discussion by doteric** [colinhacks/zod#1249] [Q&A] [answered]
Hello 👋 
Is there a way to hide the stack trace on an error during parsing and just display the issues array for example?

Thank you 😄

**Comment by nullndr** [comment, marked as answer]
You mean something like this?

```ts
const schema = z.number();

try {
  const data = schema.parse("foo");
} catch (e) {
  if (e instanceof ZodError) {
    console.log(e.issues);
  }
}
```

This will result in

```js
[
  {
    code: 'invalid_type',
    expected: 'number',
    received: 'string',
    path: [],
    message: 'Expected number, received string'
  }
]
```

**Reply by doteric** [reply]
Hey @nullndr , thanks for the reply.
Yes, something like this, but with actually throwing the error (stoping execution), not just console logging it.
So maybe `process.exit()` together with what you wrote? 🤔

**Reply by nullndr** [reply]
Yes

```ts
const schema = z.number();

try {
  const data = schema.parse("foo");
} catch (e) {
  if (e instanceof ZodError) {
    console.log(e.issues);
    process.exit(1);
  }
}
```

If you are not worried about Node stacktrace you could also throw the array:

```ts
const schema = z.number();

try {
  const data = schema.parse("foo");
} catch (e) {
  if (e instanceof ZodError) {
    throw e.issues;
  }
}
```

**Reply by doteric** [reply]
I don't want any stack traces as it needs to be fully user-friendly :D  
`process.exit` should do the job I think though. Thanks for the idea 💪
